index.ts

```diff
@@ -375,6 +385,9 @@ export default function generate(options: Options): Promise<void> {
 					if (text.charAt(0) === '.') {
 						return ` '${filenameToMid(pathUtil.join(pathUtil.dirname(sourceModuleId), text))}'`;
 					}
+					else if (options.name) {
```

Comment:
Seems simple enough, though now that this is getting more involved, perhaps we can be a bit more explicit in comments about what we're aiming to achieve with each branch here?